connections/postgres-connection/src/lib.rs

```diff
@@ -251,14 +251,26 @@ fn make_tls_connector(tls_config: &TlsConfig) -> Option<MakeTlsConnector> {
         TlsConnectMode::Disable => None,
         TlsConnectMode::Allow
         | TlsConnectMode::Prefer
-        | TlsConnectMode::Require => m_cert.and_then(|cert| {
-            let connector = TlsConnector::builder()
-                .add_root_certificate(cert)
-                .build()
-                .unwrap();
-            let connector = MakeTlsConnector::new(connector);
-            Some(connector)
-        }),
+        | TlsConnectMode::Require => {
```

Comment:
Should `Require` fall into the group with `TlsConnectMode::VerifyCa | TlsConnectMode::VerifyFull` instead? Seems like for that mode that not having a certificate is not acceptable.

Reply:
No.

`require` just requires what `prefer` prefers, or what `allow` allows (from my understanding based off https://www.postgresql.org/docs/9.1/libpq-ssl.html).

Postgres doesn't verify the server certificate (given from the server upon connection) *unless*:

1. `verify-ca` or `verify-full` are supplied, or
2. `require` is supplied WITH a root certificate (optional for this mode).

From the docs:

> Note: For backwards compatibility with earlier versions of PostgreSQL, if a root CA file exists, the behavior of sslmode=require will be the same as that of verify-ca, meaning the server certificate is validated against the CA. Relying on this behavior is discouraged, and applications that need certificate validation should always use verify-ca or verify-full.

My PR basically aligns our code to work with how postgres works (where `require` is kind of a special mode that changes its behavior based on if a root cert is present.  If a root cert is present, it *will* be used to verify the server certificates.  If a root cert is not present, *no* SSL hostname or server name validation will be done.

Reply:
Ah, ok. I get it. Thanks for the clarification!

Reply:
np! for the record and for what it's worth, i think this behavior is kind of surprising and i'm not a fan of it :p. it's caused me a lot of grief ha.